### PR TITLE
refactor: apply webpack-defaults

### LINKED ISSUE #22: webpack-defaults upgrade
Addition of `webpack-defaults` & associated refactoring as a part of the next `Major` release

Issue exists for status tracking across the organization. 

>Please do not close

### PR DESCRIPTION
**Pull request by d3viant0ne**
Intended to be merged & released as a part of `1.0.0` on a beta `dist-tag` once this has been finished and properly tested.

BREAKING CHANGE: 

Enforces NodeJS > 4.3 via engines 
Drops support for webpack < 2.x
API Change: Cachable is the default in Webpack 2.x, this option has been removed.

Closes #22

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,13 +11,58 @@
...
+    "travis:lint": "yarn run lint && yarn run security",
+    "travis:test": "yarn run test"
+  },
+  "engines": {
+    "node": ">= 4.3 < 5.0.0 || >= 5.10"
+  },
+  "main": "dist/cjs.js",
+  "files": [
+    "dist"
+  ],
```
Comment (michael-ciniawsky) [inline package.json]:
```
"repository": {
    "type": "git",
    "url": "git+https://github.com/webpack-contrib/transform-loader.git"
  },
  "bugs": {
    "url": "https://github.com/webpack-contrib/transform-loader/issues"
  },
  "homepage": "https://github.com/webpack-contrib/transform-loader",
```

File: src/lib/cacheable.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Tobias Koppers @sokra
+*/
+const loader = require('./index.js');
+module.exports = function (input) {
+  this.cacheable && this.cacheable();
```
Comment (michael-ciniawsky) [inline src/lib/cacheable.js]:
```diff
- this.cacheable && this.cacheable();
```

Maybe the whole file is obsolete, but can be triaged later...

File: package.json (hunk 1 above)
Comment (d3viant0ne) [inline package.json]: Did we change the template lately, it used to add all that stuff on init.

File: src/lib/cacheable.js (hunk 2 above)
Comment (d3viant0ne) [inline src/lib/cacheable.js]: That's what I'm looking at now, should be able to remove the entire file.

**Pull request merged**